swift_browser_ui/login.py

```diff
@@ -165,69 +173,79 @@ async def sso_query_end(
 
     try:
         # Check token availability
-        request.app['Creds'][session]['Avail'] =\
+        request.app['Sessions'][session]['Avail'] =\
             get_availability_from_token(unscoped)
     except urllib.error.HTTPError:
         raise aiohttp.web.HTTPUnauthorized(
-            reason="Token no longer valid"
+            reason="Token no longer valid",
+            headers={
+                "WWW-Authenticate": 'Bearer realm="/", charset="UTF-8"'
+            }
         )
     except urllib.error.URLError:
         raise aiohttp.web.HTTPUnauthorized(
-            reason="Cannot fetch project and domains from existing endpoint."
+            reason="Cannot fetch project and domains from existing endpoint.",
+            headers={
+                "WWW-Authenticate": 'Bearer realm="/", charset="UTF-8"'
+            }
         )
 
     if "LAST_ACTIVE" in request.cookies:
         if (request.cookies["LAST_ACTIVE"] not in [
                 p['id']
-                for p in request.app['Creds'][session]['Avail']['projects']
+                for p in request.app['Sessions'][session]['Avail']['projects']
         ]):
             raise aiohttp.web.HTTPForbidden(
                 reason="The project is not available for this token."
             )
         project_id = request.cookies["LAST_ACTIVE"]
     else:
         project_id = \
-            request.app['Creds'][session]['Avail']['projects'][0]['id']
+            request.app['Sessions'][session]['Avail']['projects'][0]['id']
 
     # Open an OS session for the first project that's found for the user.
-    request.app['Creds'][session]['OS_sess'] = initiate_os_session(
+    request.app['Sessions'][session]['OS_sess'] = initiate_os_session(
         unscoped,
         project_id
     )
 
     test_swift_endpoint(
-        request.app['Creds'][session]['OS_sess'].get_endpoint(
+        request.app['Sessions'][session]['OS_sess'].get_endpoint(
             service_type='object-store')
     )
 
     # Create the swiftclient connection
-    request.app['Creds'][session]['ST_conn'] = initiate_os_service(
-        request.app['Creds'][session]['OS_sess'],
+    request.app['Sessions'][session]['ST_conn'] = initiate_os_service(
+        request.app['Sessions'][session]['OS_sess'],
     )
 
     project_name = None
-    for i in request.app['Creds'][session]['Avail']['projects']:
+    for i in request.app['Sessions'][session]['Avail']['projects']:
         if i['id'] == project_id:
             project_name = i['name']
     # Save the current active project
-    request.app['Creds'][session]['active_project'] = {
+    request.app['Sessions'][session]['active_project'] = {
         "name": project_name,
         "id": project_id
     }
 
+    # Save time of login to the backend
+    created = time.time()
+    request.app["Sessions"][session]["last_used"] = created
+    request.app["Sessions"][session]["max_lifetime"] = created + 28800
```

Comment:
`28800` is a value that we can keep in `setd` in `settings.py` and be reused here and in `set_cookie` `max_age`